boltdesignsystem/bolt
# Fix Animated Gif Support in Bolt

## Jira
http://vjira2:8080/browse/BDS-2045

## Summary
Fixes animated Gif support in Bolt + updates the Twig to no longer expect multiple `srcset`s to be generated for Gifs.

## Details
While working on Hero examples for #1758, I noticed in one of the examples (the one that replicates [this existing Hero](https://www.pega.com/events/pegaworld)) that we have a couple of issues with how Bolt handles (or rather, _doesn't_ handle) animated Gifs:

1. The way animated Gifs were previously getting compressed / handled by our Image build task breaks existing Gif's animations

2. Our image component's Twig template expects Gifs to have multiple sizes generated vs just sticking to generating a single src / srcset.

```
<bolt-image
  src="/images/heros/hero-foreground--orange--animated-gif.gif"
  srcset="/images/heros/hero-foreground--orange--animated-gif-50.gif 50w, /images/heros/hero-foreground--orange--animated-gif-100.gif 100w, /images/heros/hero-foreground--orange--animated-gif-200.gif 200w, /images/heros/hero-foreground--orange--animated-gif-320.gif 320w, /images/heros/hero-foreground--orange--animated-gif-480.gif 480w, /images/heros/hero-foreground--orange--animated-gif.gif 550w" --> srcset isn't all that helpful here...
  max-width="450px"
  sizes="auto"
  placeholder-color="hsl(233, 33%, 97%)"
  placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
></bolt-image>;
```
^ none of these, including the original `/images/heros/hero-foreground--orange--animated-gif.gif` (after compression) continues to work

> Note: not only does this result in an assortment of Gif `srcsets` that don't animate, they also seem to flicker when switching between different screen sizes 🤦‍♂ 

## How to test

### Test 1: Confirm Gifs Now Animate
1. Pull down this branch for local dev testing
2. Download an animated Gif and move to the `docs-site/src/assets/images` folder (ex. `https://www.pega.com/sites/default/files/media/images/2019-12/pegaworld-inspire2-ani.gif`)
3. Start up the local dev environment (restart if already running)
4. Try using the animated Gif locally (via Twig) to confirm it's now working as expected (ie. animating)
```
{% include "@bolt-components-image/image.twig" with {
  src: "/images/pegaworld-inspire2-ani.gif",
} only %}
```

### Test 2: Confirm Image Cache Still Working
After running through the 1st test above:
1. Stop the local dev server
2. Delete your `www/images` folder (but nothing else!)
3. Restart the local dev server (which should regenerate assets based on the image cache, including the animated Gif)
4. Confirm the animated Gif example from the 1st test continues to work from expected (when re-generated from the image cache)
